app/views/events/_event.html.erb

```diff
@@ -1,3 +1,4 @@
+<% if event.published or can? :view_unpublished, Event %>
```

Comment:
I would say this should be filtered out in the controller, starting from the query, and not in the view.